src/main/java/de/upb/crypto/math/expressions/bool/BoolAndExpr.java

```diff
@@ -34,6 +34,27 @@ public Boolean evaluate(Substitution substitutions) {
         return lhs.evaluate(substitutions) && rhs.evaluate(substitutions);
     }
 
+    @Override
+    public LazyBoolEvaluationResult evaluateLazy(Substitution substitutions) {
+        LazyBoolEvaluationResult lhs = this.lhs.evaluateLazy(substitutions);
+        LazyBoolEvaluationResult rhs = this.rhs.evaluateLazy(substitutions);
+        if (lhs.isResultKnown())
+            return lhs.getResult() ? rhs : LazyBoolEvaluationResult.FALSE;
```

Comment:
why return rhs here?

Reply:
Because if lhs is true, then lhs ∧ rhs simplifies to rhs.

Reply:
oh the class same helps.